src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -989,14 +1022,110 @@ private T getDataObjectInstanceFromResult(FindDataObjectsResponse.Entry e) {
             return (T) dataObject;
         }
 
-
         @Override
         public Iterator<T> iterator() {
             return new ResultIterator();
         }
     }
 
     /**
+     * The page subset of data objects that matched a {@code findDataObjects} query.
+     *
+     * @param <T> data object class to be returned
+     */
+    public static class SearchPage<T extends DXDataObject> implements Iterable<T> {
```

Comment:
Can the SearchPage template parameter be made more generic so that the same class can be used for e.g. FindExecutions in the future? (Possibly, you would have to make this an abstract class and then write a concrete class that does the DXDataObject-specific transformations below.)

If the above isn't feasible for whatever reason, this should be renamed to FindDataObjectsSearchPage to avoid colliding with the similar classes we might introduce for FindExecutionsSearchPage, etc. in the future.